Update and clean up configuration files doc

noahtalerman opened the pull request:
- Clean up the top section
  - Remove section about the directory because this don't apply to someone viewing the docs on fleetdm.com
  - Remove section about how to structure the files. Instead we'll rely on great examples
  - Point user to fleetctl docs to let them know how to apply configuration files
- Clean up queries section
- Update naming for the sub-sections in the "Organization settings" section 
- Remove `single-file-configuration.yml` and files under `multi-file-configuration/`. These example files are outdated. Instead, we'll now maintain the examples in the "Configuration files" doc (README).
- Remove testing section from product handbook. This section is outdated.

chris-mcgillicuddy commented on docs/Using-Fleet/configuration-files/README.md:
```diff
@@ -7,53 +7,19 @@
 - [Organization settings](#organization-settings)
 
-Entities in Fleet, such as queries, packs, labels, agent options, and enroll secrets, can be managed with configuration files in YAML syntax.
+Fleet can be managed with configuration files (YAML syntax) and the fleetctl command-line tool.
```
```suggestion
Fleet can be managed with configuration files (YAML syntax) and the fleetctl command line tool.
```

chris-mcgillicuddy reviewed (commented):
Just one correction and it's good to go!

noahtalerman commented:
@chris-mcgillicuddy done! Thank you.

chris-mcgillicuddy commented on docs/Using-Fleet/configuration-files/README.md:
```diff
@@ -7,53 +7,17 @@
-- When defining configurations, specify the latest stable API version.
-- Configuration files should be stored in version control before being pushed to the cluster. This allows quick roll-back of a configuration if needed. It also aids with cluster re-creation and restoration if necessary.
-- Don’t specify default values unnecessarily – simple and minimal configs will reduce errors.
+Changes are applied to Fleet when the configuration file is applied using the fleetctl. Check out the [fleetctl documentation](../../Using-Fleet/fleetctl-CLI.md#using-fleetctl-to-configure-fleet) to learn how to apply configuration files.
```
```suggestion
Changes are applied to Fleet when the configuration file is applied using fleetctl. Check out the [fleetctl documentation](../../Using-Fleet/fleetctl-CLI.md#using-fleetctl-to-configure-fleet) to learn how to apply configuration files.
```

noahtalerman replied:
Great catch! Thank you.

chris-mcgillicuddy replied:
You're welcome!

chris-mcgillicuddy reviewed (commented):
Just one suggestion for flow. Sounded a bit odd to say "the fleetctl." If using "the," I'd include "CLI."

noahtalerman commented:
> Remove testing section from product handbook. This section is outdated.

@xpkoala when you get the chance, can you please confirm that this section is outdated? If I'm correct, this PR will remove this section.

chris-mcgillicuddy commented on docs/Using-Fleet/configuration-files/README.md:
```diff
@@ -447,7 +419,7 @@ For more information about integrations and Fleet automations in general, see th
 It's recommended to use the Fleet UI to configure integrations since secret credentials (in the form of an API token) must be provided. See the [Automations documentation](../../Using-Fleet/Automations.md) for the UI configuration steps.
 
-#### Organization information
+#### Org info
```
I appreciate the consistency between this and the subsequent heading, but I think "Organization information" could be easier for readers to understand — at least when we're introducing a new section. Most native English speakers will get it, but I don't know what translations of "Org info" would look like.

noahtalerman replied:
@chris-mcgillicuddy prioritizing understandability over consistency makes sense to me.

I updated my PR to use "Organization information"

chris-mcgillicuddy reviewed (commented):
@noahtalerman I just have one comment about a heading. It's more of a suggestion than a correction.

noahtalerman commented:
@chris-mcgillicuddy pinging you because this PR is now ready for rereview.

Since your last review, I added your[ latest suggestion](https://github.com/fleetdm/fleet/pull/7669#pullrequestreview-1108216233), removed several unused filed, and removed an outdated section from the product handbook.

chris-mcgillicuddy reviewed (approved):
Looks good to me!
